shotover-proxy/src/transforms/cassandra/connection.rs

```diff
@@ -108,14 +116,23 @@ async fn rx_process<C: CodecReadHalf, T: AsyncRead>(
                 match maybe_req {
                     Ok(req) => {
                         for m in req {
+                            if let Some(ref pushed_messages_tx) = pushed_messages_tx {
```

Comment:
We can use an `else if` instead of a continue here, makes the logic easier to follow.
`unwrap`ing the `as_raw_bytes` ensures we dont silently fail here if a refactor causes the message to be parsed.
Also pay attention to the formatting here, rust fmt isnt running because we are in a `select!` macro, if we wanted we could avoid that by pulling the logic into a function.
```rust
                            if let Ok(Opcode::Event) = cassandra::raw_frame::get_opcode(m.as_raw_bytes().unwrap()) {
                                if let Some(ref pushed_messages_tx) = pushed_messages_tx {
                                    pushed_messages_tx.send(vec![m]).unwrap();
                                }
                            }
                            else if let Some(stream_id) = m.stream_id() {
                                match return_channel_map.remove(&stream_id) {
                                    None => {
                                        return_message_map.insert(stream_id, m);
                                    },
                                    Some((return_tx, original)) => {
                                        return_tx.send(Response {original, response: Ok(m) })
                                            .map_err(|_| anyhow!("couldn't send message"))?;
                                    }
                                }
                            }
```

Looking at this code i'm pretty sure we previously handled events by sending them through the regular pipeline which would have caused transform invariants to break, so I dont think the cassandra_cpp client has been using events under the hood as we previously theorized.